src/flows/pipes/Columns/view.tsx

```diff
@@ -0,0 +1,126 @@
+import React, {FC, useContext, useEffect, useState} from 'react'
+import {PipeContext} from 'src/flows/context/pipe'
+import './styles.scss'
+import {PipeProp} from 'src/types/flows'
+import {
+  ResourceCard,
+  SlideToggle,
+  FlexBox,
+  InputLabel,
+  FlexDirection,
+  AlignItems,
+  ComponentSize,
+} from '@influxdata/clockface'
+
+export interface TableColumnKey {
+  name: string
+  visible: boolean
+}
+const View: FC<PipeProp> = ({Context}) => {
+  const [tableColumnKeys, setTableColumnKeys] = useState<{
+    [_: string]: TableColumnKey
+  }>({})
+
+  const {update, data, results} = useContext(PipeContext)
+
+  useEffect(() => {
+    if (results?.parsed?.table) {
+      setTableColumnKeys(
+        results.parsed.table.columnKeys.reduce((a, b) => {
+          if (!data.updatedTableKeys[b]) {
+            a[b] = {name: b, visible: true}
+          } else {
+            a[b] = data.updatedTableKeys[b]
+          }
+          return a
+        }, {})
+      )
+    }
+  }, [results?.parsed?.table])
+
+  useEffect(() => {
+    // const updatedKeys = Object.entries(tableColumnKeys).reduce((a, [k, v]) => {
+    //   if (v.name === k) {
+    //     return a
+    //   } else {
+    //     return {...a, [k]: {...v}}
+    //   }
+    // }, {})
+
+    update({
+      updatedTableKeys: tableColumnKeys,
+    })
+  }, [tableColumnKeys])
+
+  const handleUpdateName = (newName: any, oldName: string) => {
+    if (tableColumnKeys[oldName].name === newName) {
+      return
+    }
+    setTableColumnKeys(prev => ({
+      ...prev,
+      [oldName]: {
+        ...tableColumnKeys[oldName],
+        name: newName,
+      },
+    }))
+  }
+
+  const handleUpdateVisible = (oldName: string, newVisibleStatus: boolean) => {
+    if (tableColumnKeys[oldName].visible === newVisibleStatus) {
+      return
+    }
+
+    setTableColumnKeys(prev => ({
+      ...prev,
+      [oldName]: {
+        ...tableColumnKeys[oldName],
+        visible: newVisibleStatus,
```

Comment:
connected to the issue below, why not just use the previous state to set the visibility to the opposite of what it was?

```
visible: !prev[oldName].visible
```

or something like that